src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -466,13 +464,9 @@ public void saveProgram(File f) throws IOException {
             throw new IOException("Failed to persist "+f,e);
         } finally {
             PROGRAM_STATE_SERIALIZATION.set(old);
+            execution.time(TimingKind.save, true);
             Util.deleteFile(tmpFile);
         }
-
-        if (logging) {
```

Comment:
Although... potentially that level of fine grain might be better to reserve for OS-level or profiler instrumentation.